src/components/select/select.jsx

```diff
@@ -121,13 +123,23 @@ const Select = forwardRef(function Select(props, ref) {
   useEffect(() => {
     if (!mounted.current) return;
 
-    validate();
+    if (hasBeenBlurred) {
+      validate();
+    }
 
     if (props.value === value) return;
 
     props.onChange({ target: selfRef.current });
   }, [value]);
 
+  function handleBlur(event) {
+    if (!(listBoxRef.current && listBoxRef.current.contains(event.relatedTarget))) {
```

Comment:
`FocusEvent.relatedTarget` is experimental. But I think it's okay to keep this implementation for now. I think @Bloodyaugust found a React-only solution but it requires upgrading our React version or something. If we do find problems with `relatedTarget` then we can just focus big-time on a React specific solution.

![image](https://user-images.githubusercontent.com/841084/114468007-a9742a80-9b9f-11eb-8508-4eabd499be8c.png)